Change migration name to avoid conflicts

arpitchauhan opened the pull request:
The migration AddMissingIndexes conflicts with a migration of similar
name in other gems, eg acts_as_taggable_on 4.0.0. Its name has been
made more specific in order to avoid such conflicts while installing
migrations in a rails app using

```
bundle exec rake railties:install:migrations
```

krzysiek1507 commented:
Hi @arpitchauhan could you please rebase branch and create PR for `master` too?

arpitchauhan commented:
@krzysiek1507 Rebasing is proving quite complicated. Is it ok if I cherry-pick my commit onto a new branch made out of master?

krzysiek1507 commented:
@arpitchauhan sorry. I mean, rebase your branch with `3-1-stable` and also create new PR to `master` with cherry-picked commits. ;) I hope now it's clear.

arpitchauhan commented:
Sure, will do.
On Fri, Aug 12, 2016 at 10:12 PM Krzysztof Rybka <notifications@github.com>
wrote:

> @arpitchauhan <https://github.com/arpitchauhan> sorry. I mean, rebase
> your branch with 3-1-stable and also create new PR to master with
> cherry-picked commits. ;) I hope now it's clear.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/spree/spree/pull/7526#issuecomment-239496915>, or mute
> the thread
> <https://github.com/notifications/unsubscribe-auth/ACQnT9dv5jh7twoVJfeQ66ctUMM-C996ks5qfKKLgaJpZM4Ji7lu>
> .
>

arpitchauhan commented:
@krzysiek1507 Done. The other PR is #7530

krzysiek1507 commented:
Hi @arpitchauhan could you add checking if indexes exist?

arpitchauhan commented:
@krzysiek1507 do you mean adding spec examples (tests)?

arpitchauhan commented:
@krzysiek1507 Ah, got you. Will add them.

arpitchauhan commented:
@krzysiek1507 done

krzysiek1507 commented:
@arpitchauhan could you squash commits please?

Commit:
Change migration name to avoid conflicts

The migration AddMissingIndexes conflicts with a migration of
similar name with other gems, eg acts_as_taggable_on 4.0.0. Its name has
been made more specific in order to avoid such conflicts while
installing migrations in a rails app.

Also, since the migration will be copied over and run again, checks have
been inserted ensuring the indexes being added don't exist already.

arpitchauhan commented:
@krzysiek1507 done

krzysiek1507 commented on core/db/migrate/20150314013438_add_missing_indexes_on_spree_tables.rb:
```diff
@@ -0,0 +1,57 @@
+class AddMissingIndexesOnSpreeTables < ActiveRecord::Migration
+  def change
+    unless index_exists?(:spree_promotion_rules_users, [:user_id, :promotion_rule_id])
+      add_index   :spree_promotion_rules_users,
```
Shouldn't we consider index name in `index_exists?`? And please fix indentation.

arpitchauhan replied:
@krzysiek1507 The [documentation](http://apidock.com/rails/ActiveRecord/ConnectionAdapters/SchemaStatements/index_exists%3F) says we need to use table name and column name(s) as arguments.

I'll fix the spacing between `add_index` and arguments.

arpitchauhan replied:
@krzysiek1507  done

krzysiek1507 replied:
@arpitchauhan Yes, but we can put `name` into `options` hash (4th example).
If you check source, you will see that name will be consider too.

arpitchauhan replied:
@krzysiek1507 got it, thanks. Done.

arpitchauhan commented:
@krzysiek1507 is it possible for this PR to be included in 3.1.2?

arpitchauhan commented:
@damianlegawiec Similar MR for master branch was merged ( #7530 ). Can this be merged too?

damianlegawiec commented:
Sure @arpitchauhan  and thanks for the fix!
